app/controllers/devise_token_auth/concerns/resource_finder.rb

```diff
@@ -20,7 +20,9 @@ def get_case_insensitive_field_from_resource_params(field)
   end
 
   def find_resource(field, value)
-    @resource = if resource_class.try(:connection_config).try(:[], :adapter).try(:include?, 'mysql')
+    @resource = if :email == field && Gem.loaded_specs[ 'devise-multi_email' ]
```

Comment:
This is ugly, can't you check if a class defined in that class exists or something?

Reply:
Is this better?
```suggestion
    @resource = if :email == field && Object.const_defined?( 'Devise::MultiEmail' )
```

I can't test it fully in my environment at the moment because of the Devise/OmniAuth issue, but it seems to work under minimal testing.

Reply:
I've got an update coming along with the beginnings of tests. I'm going to post it here shortly and I hope you'll have a look. I'm not sure how to work around this [devise/omniauth](https://github.com/heartcombo/devise/issues/5326) issue yet, but I'm working on it...